node/src/tasks/user_sends_file.rs

```diff
@@ -182,15 +184,16 @@ where
             info!(target: LOG_TARGET, "No peers were found to receive file key {:?}", file_key);
         }
 
-        self.send_chunks_to_provider(peer_ids, &file_metadata).await
+        self.send_chunks_to_providers(peer_ids, &file_metadata)
+            .await
     }
 }
 
 impl<NT> UserSendsFileTask<NT>
 where
     NT: ShNodeType,
 {
-    async fn send_chunks_to_provider(
+    async fn send_chunks_to_providers(
```

Comment:
```suggestion
    async fn send_chunks_to_provider(
```

Actually it is still sending it to just ONE provider, which can have many peers.